contrib/spring_boot_actuator/src/main/java/io/opentelemetry/contrib/spring/boot/actuate/MeterSdkRegistryBean.java

```diff
@@ -0,0 +1,120 @@
+
+import io.opentelemetry.metrics.MeterRegistry;
+import io.opentelemetry.sdk.common.Clock;
+import io.opentelemetry.sdk.internal.MillisClock;
+import io.opentelemetry.sdk.metrics.MeterSdkRegistry;
+import io.opentelemetry.sdk.metrics.MeterSdkRegistryProvider;
+import io.opentelemetry.sdk.resources.EnvVarResource;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Creates an OpenTelemetry {@link MeterRegistry} from the default SDK the Spring way. */
+@Component
+public class MeterSdkRegistryBean implements FactoryBean<MeterRegistry>, InitializingBean {
+
+  private OpenTelemetryProperties properties;
+  private Clock clock;
+  private Resource resource;
+  private MeterRegistry meterRegistry;
+
+  /**
+   * Sets the Spring application properties used to configure the OpenTelemetry SDK.
+   *
+   * @param properties the properties
+   */
+  @Autowired
+  public void setProperties(OpenTelemetryProperties properties) {
+    this.properties = properties;
+  }
+
+  /**
+   * Sets the Spring-managed OpenTelemetry clock implementation for generating span timestamps.
+   *
+   * @param clock the clock
+   */
+  @Autowired(required = false)
+  public void setClock(Clock clock) {
+    this.clock = clock;
+  }
+
+  /**
+   * Sets the Spring-managed implementation of OpenTelemetry {@link Resource} which populates global
+   * application attributes.
+   *
+   * @param resource the resource populator
+   */
+  @Autowired(required = false)
+  public void setResource(Resource resource) {
+    this.resource = resource;
+  }
+
+  @Override
+  public void afterPropertiesSet() {
+    meterRegistry = initializeMeterRegistry();
+  }
+
+  @Override
+  public MeterRegistry getObject() {
+    if (meterRegistry == null) {
+      meterRegistry = initializeMeterRegistry();
+    }
+    return meterRegistry;
+  }
+
+  @Override
+  public Class<?> getObjectType() {
+    return MeterRegistry.class;
+  }
+
+  private MeterRegistry initializeMeterRegistry() {
+    MeterSdkRegistryProvider provider = new MeterSdkRegistryProvider();
+    MeterSdkRegistry registry = (MeterSdkRegistry) provider.create();
+    if (properties.getMeter().isExport()) {
+      constructMeterSharedState();
+    }
+    return registry;
+  }
+
+  private Object constructMeterSharedState() {
+    Map<String, Object> state = new HashMap<>();
+    state.put("clock", prepareClock());
+    state.put("resource", prepareResource());
+    return state;
+  }
+
+  private Clock prepareClock() {
+    if (clock != null) {
+      return clock;
+    }
+    return MillisClock.getInstance();
+  }
+
+  private Resource prepareResource() {
+    if (resource != null) {
+      return resource;
+    }
+    return EnvVarResource.getResource();
```

Comment:
I wonder if this is really how it would work best in Spring. I haven't written production spring code in a while, but I would suspect that something other than an env var would be more idiomatic?

Reply:
This is a good point. I will create a Spring Boot Resource that uses the auto-configuration mechanism to extract the appropriate values.